### PR TITLE
Fix hidden value domain whitelist

### PR DESCRIPTION
**Pull request by marcoquerque**
A few issues:
 - if you tried to change value_is_hidden - it wasn't properly destroying and recreating
 - You should only ever try to set HiddenValueDomainWhitelist when userAttributeValueIsHidden and it is none empty.

### CODE REVIEW

**Comment by marcoquerque** [comment]
@hirosassa do you want me to include the version bumps in this PR?

**Comment by hirosassa** [comment]
@marcoquerque Thanks for your contribution! Yes, please bump the version 👍

**Comment by marcoquerque** [comment]
@hirosassa done!

File: pkg/looker/resource_user_attribute.go [hunk 1]
```diff
@@ -76,7 +77,12 @@ func resourceUserAttributeCreate(ctx context.Context, d *schema.ResourceData, m
...
 		UserCanView:                &userAttributeUserCanView,
 		UserCanEdit:                &userAttributeUserCanEdit,
-		HiddenValueDomainWhitelist: &userAttributeHiddenValueDomainWhitelist,
+	}
+
+	if userAttributeValueIsHidden && userAttributeHiddenValueDomainWhitelist != "" {
+
+		writeUserAttribute.HiddenValueDomainWhitelist = &userAttributeHiddenValueDomainWhitelist
+
 	}
```
Comment (hirosassa) [inline pkg/looker/resource_user_attribute.go:86]: Could you remove redundant empty lines like below?

```suggestion
	if userAttributeValueIsHidden && userAttributeHiddenValueDomainWhitelist != "" {
		writeUserAttribute.HiddenValueDomainWhitelist = &userAttributeHiddenValueDomainWhitelist
	}
```

File: pkg/looker/resource_user_attribute.go [hunk 2]
```diff
@@ -152,7 +158,12 @@ func resourceUserAttributeUpdate(ctx context.Context, d *schema.ResourceData, m
 		ValueIsHidden:              &userAttributeValueIsHidden,
 		UserCanView:                &userAttributeUserCanView,
 		UserCanEdit:                &userAttributeUserCanEdit,
-		HiddenValueDomainWhitelist: &userAttributeHiddenValueDomainWhitelist,
+	}
+
+	if userAttributeValueIsHidden && userAttributeHiddenValueDomainWhitelist != "" {
+
+		writeUserAttribute.HiddenValueDomainWhitelist = &userAttributeHiddenValueDomainWhitelist
+
```
Comment (hirosassa) [inline pkg/looker/resource_user_attribute.go:166]: Same as here 
```suggestion
	}

	if userAttributeValueIsHidden && userAttributeHiddenValueDomainWhitelist != "" {
		writeUserAttribute.HiddenValueDomainWhitelist = &userAttributeHiddenValueDomainWhitelist
```

**Review by hirosassa — CHANGES_REQUESTED** [review]
Thanks! Left a few small comments.

File: pkg/looker/resource_user_attribute.go (hunk 1 above)
Comment (marcoquerque) [reply pkg/looker/resource_user_attribute.go:86]: Is this just go formatting?

Comment (marcoquerque) [reply pkg/looker/resource_user_attribute.go]: oh i see just new lines - :+1:

**Review by hirosassa — APPROVED** [review]
LGTM!

**Comment by hirosassa** [comment]
@marcoquerque released!

https://registry.terraform.io/providers/hirosassa/looker/0.8.8

**Pull request merged**